genai/example_test.go

```diff
@@ -1119,6 +1119,33 @@ func ExampleClient_UploadFile_image() {
 
 }
 
+func ExampleClient_UploadFile_pdf() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	file, err := uploadFile(ctx, client, filepath.Join(testDataDir, "test.pdf"), "")
```

Comment:
I thought we should inline `uploadFile`?

Reply:
They only asked to inline it for videos.

Inlining it everywhere will blow up these examples and duplication even more. I think the right thing to do is add a convenience API to the client itself that implements this functionality (reads from a path, not a reader), this will enable us to remove the helper function

Reply:
See https://github.com/google/generative-ai-go/pull/203 for a sketch